### ISSUE TITLE
Threading speedup not available with ffmpeg2vmaf

### ISSUE THREAD

**Issue by nandakd** [Netflix/vmaf#197]
Is the speed improvement with threads=0 only available for the vmafossexec wrapper? It's not available with ffmpeg2vmaf?

**Comment by li-zhi** [comment]
Unfortunately it is not yet available. It involves migrating the multithreading logic from `wrapper/src/combo.c` to `feature/src/all.c`.

An alternative approach is to modify the script `ffmpeg2vmaf` to replace `VmafQualityRunner` class with `VmafossExecQualityRunner`. This way, it pipes the FFmpeg output to `vmafossexec`.

**Comment by DonTequila** [comment]
If you are on Linux you can take a look at my fork. I added reading the source frames from a ffmpeg pipe into the vmafossexec wrapper. If this is interesting I can make it more generic and create a pull request (had asked li-zhi before). I don't think it will work on Windows though because I couldn't find a way to get popen on Linux.
https://github.com/DonTequila/vmaf

**Comment by DonTequila** [comment]
If you are on Linux you can take a look at my fork. I added reading the source frames from a ffmpeg pipe into the vmafossexec wrapper. If this is interesting I can make it more generic and create a pull request (had asked li-zhi before). I don't think it will work on Windows though because I couldn't find a way to get popen on Windows.
https://github.com/DonTequila/vmaf

**Comment by li-zhi** [comment]
@DonTequila yes that will be helpful. Please feel free to create a pull request.

**Comment by nandakd** [comment]
@DonTequila and @li-zhi , thanks - that will be very helpful.

**Comment by cosmin** [comment]
At this point I would recommend switching to `ffmpeg` with `libvmaf` (if you have the option to compile your own ffmpeg) rather than using `ffmpeg2vmaf`, because this does take advantage of multi-threading which makes a big difference, or if not then wrapping `vmafossexec` in ffmpeg pipes. It's hard to imagine why anyone would use `ffmpeg2vmaf` at all at this point.

**Comment by li-zhi** [comment]
@cosmin I agree. The ffmpeg with libvmaf filter should be the paved path forward.
